geojson.go

```diff
@@ -0,0 +1,175 @@
+package simplefeatures
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math"
+	"strconv"
+)
+
+func UnmarshalGeoJSON(r io.Reader) (Geometry, error) {
+	var firstPass struct {
+		Type string `json:"type"`
+	}
+	var rCopy bytes.Buffer
+	if err := json.NewDecoder(io.TeeReader(r, &rCopy)).Decode(&firstPass); err != nil {
```

Comment:
The copy into the `bytes.Buffer` wouldn't be necessary if I were to pass in a `[]byte` instead of an `io.Reader`... I'll do that instead.

Reply:
Fixed.